Trisha-tech/OnlineBookSales
# Added the context  and redux for cart,wishlist,searchBar,pricefilter.

### Description

- This PR does the following:

  - Adds context for search functionality and price filtering.
  - Implements Redux for managing shopping cart and wishlist states.
  - Updates the navbar with wishlist and cart icons displaying item counts.
  - fixing the bug of login especially when someone refresh the page.
  - create the cart page,wishlist page and styled it.
  - connected frontend with backend
  - make the profile page and connect it with backend showing the data.
  - fixing some of the things in backend .
  - ensure everything working properly
  - added loader to show while fetching from database
  - made some of the page that is required
  -add the filter for category

###   **- Now i have connectedd the frontend with backend and everything working properly i have save some random data to display the things ,you have to save something to view it .**
![Screenshot 2024-05-31 085335](https://github.com/Trisha-tech/OnlineBookSales/assets/136432304/c09830cd-4d28-41ae-b015-125989df698e)
![Screenshot 2024-05-31 085444](https://github.com/Trisha-tech/OnlineBookSales/assets/136432304/80b6a09f-f0bf-41db-a61e-4067b7a1e21f)


### I  have tested everything it is working fine and have added context related to filter books on the basis of types of book.

### Related Issues
fixes #135 

### Changes
List the detailed changes made in this PR.
- Added `SearchBarContext` for managing search functionality.
- Added `PriceFilterContext` for managing price filtering.
- Integrated Redux to manage the shopping cart and wishlist states.
- Updated `Shop` component to utilize contexts for filtering products.
- Styled `PriceFilter` component for improved user interface.
- Updated the navbar with wishlist and cart icons showing item counts and added hover effects.

### Testing Instructions
Detailed instructions on how to test the changes. Include any setup needed and specific test cases.
1. Pull this branch.
2. Run `npm install` to install dependencies.
3. Run `npm test` to execute the test suite.
4. Verify that the search and price filtering functionalities work as expected.
5. Check that the wishlist and cart icons in the navbar display the correct item counts and have hover effects.
6. Ensure that the shopping cart and wishlist states are managed correctly using Redux.

### Screenshots (if applicable)
Add any screenshots that help explain or visualize the changes.

![Screenshot 2024-05-31 090612](https://github.com/Trisha-tech/OnlineBookSales/assets/136432304/0d1aea5e-ca1f-44b3-a942-4a7bc1a11ccd)
this is the filter button will aplly on shop page while shopping.


### Additional Context
Any additional context or information that reviewers should be aware of.
- This PR is based on the implementation of context and Redux to manage state more efficiently.

### Checklist
Make sure to check off all the items before submitting. Mark with [x] if done.
